spec/Section 3 -- Type System.md

```diff
@@ -1771,18 +1771,82 @@ This allows inputs which accept one or many arguments (sometimes referred to as
 single value, a client can just pass that value directly rather than
 constructing the list.
 
+The result of coercion of a value {value} to a list type {listType} is
+{CoerceListValue(value, listType)}.
+
+CoerceListValue(value, listType):
+
+- If {value} is {null}, return {null}.
+- Let {itemType} be the inner type of {listType}.
+- Let {coercedList} be an empty list.
+- If {value} is a list:
+  - For each {itemValue} in {value}:
+    - Let {coercedItemValue} be {CoerceListItemValue(itemValue, itemType)}.
+    - Append {coercedItemValue} to {coercedList}.
+- Otherwise:
+  - Let {coercedItemValue} be {CoerceListItemValue(value, itemType)}.
+  - Append {coercedItemValue} to {coercedList}.
+- Return {coercedList}.
+
+CoerceListItemValue(itemValue, itemType):
+
+- If {itemValue} is {null}, return {null}.
+- Otherwise, if {itemValue} is a Variable:
+  - If the variable provides a runtime value:
+    - Let {coercedItemValue} be the runtime value of the variable.
+  - Otherwise, if the variable definition provides a default value:
+    - Let {coercedItemValue} be this default value.
```

Comment:
Is there a specific reason this doesn't use the "pre-coerced" variable values?

Reply:
The reason was that this is essentially copied from the input coercion for input objects (but via an algorithm to make it clearer): https://spec.graphql.org/draft/#sec-Input-Objects.Input-Coercion

But it's a good question. I guess the reason is that `coercedVariableValues` is not explicitly made available in section 3 of the spec.

Reply:
I see... It may be an issue:

Ultra synthetic example:

```graphql
type Query {
  a(b: [[Int]]): Int
}
```

Operation:

```graphql
query Foo($c: [Int]) {
  a(b: [$c])
}
```

Runtime Variables:
```json
{
  "c": 42
}
```

If we're saying the runtime value is the "value that is sent over the wire", we end up with `b = [42]` (incompatible) instead of `b = [[42]]` if we coerce the variable to a list first (I think?)
If we're saying the runtime value is the "pre-coerced" value, then there's no need to mention defaultValue